File: backend/app/portfolio/portfolio_service.py
from app.auth.auth_service import auth_service


class PortfolioService:
    """
    Service for retrieving portfolio-related information.
    """

    def get_holdings(self):
        response = auth_service.client.get_holdings()
        return response.json()

    def get_positions(self):
        response = auth_service.client.get_net_position()
        return response.json()

    def get_funds(self):
        response = auth_service.client.get_fund_summary()
        return response.json()


portfolio_service = PortfolioService()

File: frontend/src/services/portfolioService.ts
const API_URL = "http://127.0.0.1:8000";

export async function getPositions() {
  const response = await fetch(`${API_URL}/portfolio/positions`);

  if (!response.ok) {
    throw new Error("Failed to fetch positions");
  }

  return response.json();
}

export async function getHoldings() {
  const response = await fetch(`${API_URL}/portfolio/holdings`);

  if (!response.ok) {
    throw new Error("Failed to fetch holdings");
  }

  return response.json();
}

export async function getFunds() {
  const response = await fetch(`${API_URL}/portfolio/funds`);

  if (!response.ok) {
    throw new Error("Failed to fetch funds");
  }

  return response.json();
}